### PR TITLE
DDS-1154 CI/CD Support, Testing and SCSS

### PR DESCRIPTION
**Pull request by yodas7**
JIRA: https://daikinsvoil.atlassian.net/browse/DDS-1154

- Introduce Prettier and EditorConfig with the same config as component library
- Migrate to TypeScript
- Update dependencies
- Publish JSON files
- Add SCSS mixins
  - This will probably be necessary for future dark theme support.
- Add tests
  - To ensure ESM and CJS builds actually work.
- Add CD powered by Changesets
  - requires `NPM_TOKEN` secret.
  - demo: yodas7/dds-tokens#2
- Add Spell Checker (only on CI)
  - If a misspelling is detected, it will be displayed on the PR, but the CI will not fail. 

I'm concerned that the theme folder contains capital letters (`Dark` and `Light`).
I think it's okay because users use bundler, but the component library is unified in lowercase, so I thought it would be okay to unify it in lowercase as well. However, this would be a breaking change.

### Checklist

- [x] Add `NPM_TOKEN` secret (@hiro-daikin).

### CODE REVIEW

**Comment by yodas7** [comment]
BTW I found a typo:
https://github.com/dsv-rp/dds-tokens/blob/main/themes/system.json#L402

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -1,19 +1,26 @@
-name: Build tokens
+name: Build Tokens
 
 on:
-  workflow_dispatch:
-  workflow_call:
```
Comment (hiro-daikin) [inline .github/workflows/build.yaml:5]: We might still want the ability to test the build manually on specific branches

File: scripts/lib/compare.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    text += `**BREAKING CHANGE**: ${result.totalRemoved} token(s) removed.  \n`;
+  }
+  if (result.totalAdded > 0) {
+    text += `${result.totalAdded} token(s) added.  \n`;
+  }
+  if (result.totalModified > 0) {
+    text += `${result.totalModified} token(s) updated.  \n`;
+  }
+  return text;
+}
```
Comment (hiro-daikin) [inline scripts/lib/compare.ts:96]: Do we plan on writing unit tests for these?

**Review by hiro-daikin — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yaml (hunk 1 above)
Comment (yodas7) [reply .github/workflows/build.yaml:5]: Reverted. Thanks!

File: scripts/lib/compare.ts (hunk 2 above)
Comment (yodas7) [reply scripts/lib/compare.ts:96]: Added tests for lib/*.ts files.

**Comment by yodas7** [comment]
@hiro-daikin 
I will remove the draft status once the NPM_TOKEN secret has been added to the repository, so please let me know when it has been added.

**Comment by hiro-daikin** [comment]
@yodas7 Please let me know on slack once you remove the `draft` status

**Comment by yodas7** [comment]
@hiro-daikin 
I may have made a mistake with the English.
I don't have an npm account that I can use at work, meaning I can't add NPM_TOKEN, so I would like to ask hiro-san to add it.

**Comment by hiro-daikin** [comment]
@yodas7 I may have misunderstood - my apologies.
I just added the secret: `NPM_TOKEN`. 
It should be the same as in the DDS repository

**Review by hiro-daikin — DISMISSED** [review]

**Comment by yodas7** [comment]
@hiro-daikin 
Thank you! I have confirmed that the token has been added using the `gh secret list` command.
I will ask someone in my company to review it, and if there are no problems, I will merge it.

**Review by ukkary — DISMISSED** [review]

**Comment by yodas7** [comment]
I dismissed the review. I will self-approve it.

**Review by poetrainy — APPROVED** [review]

**Pull request merged**